scripts/o3de/o3de/utils.py

```diff
@@ -248,7 +248,7 @@ def download_progress(downloaded_bytes):
     return 0
 
 
-def download_zip_file(parsed_uri, download_zip_path: pathlib.Path, force_overwrite: bool, object_name = str, download_progress_callback = None) -> int:
+def download_zip_file(parsed_uri, download_zip_path: pathlib.Path, force_overwrite: bool, object_name: str, download_progress_callback = None) -> int:
```

Comment:
The two changes in this file look the same in the old version but they ended up different in the new version; one has a default value and one doesn't.

Reply:
Yup, that was the bug, `= str` was assigning a default value of `str` which is the Python string type instead of `: str` which indicates the parameter type should be `str` with no default value.

The reason I didn't provide a default parameter for `download_zip_file` is there is only one usage of it whichpasses in the object_name, but `download_file` does have usages where the object_name param is optional.  We could make it optional, but I went with the more restrictive option for this internal utility function.